server/api/db/sqldb/db.py

```diff
@@ -1856,7 +1880,7 @@ def get_function(
 
     def delete_function(self, session: Session, project: str, name: str):
         logger.debug("Removing function from db", project=project, name=name)
-
+        self._delete(session, FunctionSpecFoo, project=project)
```

Comment:
I don't see a reason to delete them separately, they should be cascaded with the functions.
You could remove dangling once like we do with labels below.